### ISSUE TITLE
A slimmer python

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/python-feedstock#581]
### Comment:

I'm wondering if there is enough enthusiasm for having a slim python build. My main motivation is to help ensure that downstream packages that depend on python, also declare the dependencies they have on other libraries. The ones that come to mind are:

- [ ] readline
- [ ] sqlite
- [ ] tk

I'm not sure about the implementation, but maybe we can have a behavior similar to pip, where during the build process they aren't pulled in automatically, but for regular users they are? I know that the pip feature is engrained pretty deep into conda, so maybe it isn't that desirable to use it, but the behavior:
- At build time, maintainers might have to specify `python-readline`, `python-sqlite`, `python-tk` as part of their dependencies.
- At environment making time, maybe those are all pulled automatically?

With non-python packages, this has typically resulted in the creation of a "core" package. This would mean that the dependency graph might look like
````
           /----- python-readline ----\------------------> readline
          /------ python-sqlite -------\-----------------> sqlite
python ------------------------------------ python-core
          \-------python-tk------------/-----------------> tk
```
with `tk`, `sqlite` and `readline` being removed as dependencies from `python-core`.


PS. I know this conversation is quite cyclic in nature but I figured it is time to revive it.

**Comment by jakirkham** [comment]
Yeah it sounds like the same thing. Let's consolidate into this issue. If you spot anything else that hasn't already been suggested for splitting above, please feel free to add that as a comment here. Thanks! 🙏

**Comment by peterroelants** [comment]
From Duplicate: https://github.com/conda-forge/python-feedstock/issues/586

### Comment:

Would it be of interest to provide a `python-minimal` package in conda-forge providing a minimal installation of Python, similar in spirit to e.g. the Debian/Ubuntu [`python3.10-minimal`](https://packages.ubuntu.com/jammy/python3.10-minimal) packages?

I noticed that the python3.10 lib directories differ about 50MB in size:
- `apt install python3.10-minimal`: `22M	/usr/lib/python3.10/`
- `micromamba create -n py310 --channel conda-forge python=3.10.*`: `70M	/root/micromamba/envs/py310/lib/python3.10`

The conda-forge `python` package contains things like `turtledemo`, `tkinter`, `lib2to3` which are not typical requirements and are not in `python3.10-minimal`.

**Comment by jakirkham** [comment]
Do `turtledemo` or `lib2to3` take up significant space?

**Comment by hmaarrfk** [comment]
about 0.5MB and 1.2MB respectively it seems.

**Comment by jakirkham** [comment]
IMHO `lib2to3` still feels pretty core to Python. Some packages rely on it in their build processes.

No strong feelings on `turtledemo`.
